pkg/kubelet/volumes.go

```diff
@@ -143,7 +143,8 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 			return nil, err
 		}
 		if attacher != nil {
-			err = attacher.Attach(volSpec, kl.hostname)
+			host := &volumeHost{kl}
+			err = attacher.Attach(host, volSpec, kl.hostname)
```

Comment:
Attach/Detach should only return error. The attach/detach controller, for example, doesn't care where the device is mounted. Instead the `WaitForAttach`/`WaitForDetach` methods should return that info.